src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,163 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(tail, value, null);
+        if (head == null) {
+            head = tail = node;
+        } else {
+            tail.next = node;
+            node.prev = tail;
+            tail = node;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index is out of bounds " + index);
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index == 0) {
+            addFirst(value);
+        } else {
+            Node<T> current = getNode(index);
+            Node<T> newNode = new Node<>(current.prev.next,value,current);
+            current.prev.next = newNode;
+            current.prev = newNode;
+            size++;
+        }
```

Comment:
```suggestion
        if (index == 0) {
            addFirst(value);
            return;
       }
        Node<T> current = getNode(index);
        Node<T> newNode = new Node<>(current.prev.next,value,current);
        current.prev.next = newNode;
        current.prev = newNode;
        size++;
        }
```
remember, wrapped/indented code is harder to read, in situations like this, it's better to return early